java/ql/lib/semmle/code/java/Statement.qll

```diff
@@ -428,6 +450,20 @@ class SwitchCase extends Stmt, @case {
     result = this.getSwitch().getExpr() or result = this.getSwitchExpr().getExpr()
   }
 
+  /**
+   * Gets the `i`th case in this case's switch block.
+   */
+  SwitchCase getSiblingCase(int i) {
+    result = this.getSwitch().getCase(i) or result = this.getSwitchExpr().getCase(i)
+  }
+
+  /**
+   * Gets this case's ordinal in its switch block.
+   */
+  int getCaseIndex() {
+    this = this.getSwitch().getCase(result) or this = this.getSwitchExpr().getCase(result)
```

Comment:
There's actually a superfluous join here - not that I think it matters much, but we could simplify a bit:
```suggestion
    this = any(SwitchStmt s).getCase(result) or this = any(SwitchExpr s).getCase(result)
```